vbfml/scripts/train.py

```diff
@@ -87,38 +78,69 @@ def setup(ctx, learning_rate: float, dropout: float):
         if re.match(dataset_labels["v_qcd_nlo_17"], dataset_info.name):
             dataset_info.n_events = 0.01 * dataset_info.n_events
 
+    # Object containing data for different models
+    # (set of features, dropout rate etc.)
+    # Loaded from the YML configuration file
+    ymlfile = "../config/model_params.yml"
+    db = ModelDB(ymlfile)
+    db.set_model(model)
+
+    features = db.get("features")
+
     training_sequence = build_sequence(
-        datasets=copy.deepcopy(datasets), features=features
+        datasets=copy.deepcopy(datasets),
+        features=features,
+        weight_expression=db.get("weight_expression"),
     )
     validation_sequence = build_sequence(
-        datasets=copy.deepcopy(datasets), features=features
+        datasets=copy.deepcopy(datasets),
+        features=features,
+        weight_expression=db.get("weight_expression"),
     )
     normalize_classes(training_sequence)
     normalize_classes(validation_sequence)
+
     # Training sequence
-    training_sequence.read_range = (0.0, 0.5)
+    train_size = db.get("train_size")
+    training_sequence.read_range = (0.0, train_size)
     training_sequence.scale_features = True
-    training_sequence.batch_size = 20
-    training_sequence.batch_buffer_size = 1e6
+    training_sequence.batch_size = db.get("batch_size_train")
+    training_sequence.batch_buffer_size = db.get("batch_buffer_size_train")
     training_sequence[0]
 
     # Validation sequence
-    validation_sequence.read_range = (0.5, 1.0)
+    validation_sequence.read_range = (train_size, 1.0)
     validation_sequence.scale_features = True
     validation_sequence._feature_scaler = copy.deepcopy(
         training_sequence._feature_scaler
     )
-    validation_sequence.batch_size = 1e6
-    validation_sequence.batch_buffer_size = 10
+    validation_sequence.batch_size = db.get("batch_size_val")
+    validation_sequence.batch_buffer_size = db.get("batch_buffer_size_val")
 
     # Build model
-    model = sequential_dense_model(
-        n_layers=3,
-        n_nodes=[4, 4, 2],
-        n_features=len(features),
-        n_classes=len(training_sequence.dataset_labels()),
-        dropout=dropout,
-    )
+    if model == "dense":
```

Comment:
let's put all of this in the config file and put the creation code in a factory so that I can just call:
```
model_config = ModelConfiguration("/path/to/file.yml")
model = ModelFactory.build(model_config)
```

(which ModelFactory.build being a class method)

Reply:
This is implemented in 676a7a3